### DISCUSSION TITLE
Any plan to fix strictMode breaking source maps in metro-serializer-esbuild?

### DISCUSSION THREAD

**Discussion by BrentonWheeler** [microsoft/rnx-kit#2754] [Q&A] [answered]
Hey thanks for the great repo, just wanted to check if there are any plans to fix the issue of the `strictMode: false` option breaking source maps in `metro-serializer-esbuild`?

https://microsoft.github.io/rnx-kit/docs/tools/metro-serializer-esbuild#strictmode

> Note that disabling strictMode here will definitely break source maps. It is recommended to try disabling strict mode in Babel or TypeScript first before considering this option. If you can target ES6, that is a better alternative.

Unfortunately I haven't been able to find an alternative solution to using `strictMode: false` to my specific issue, which is very similar to the one found here: https://github.com/callstack/haul/issues/278 - which results in android crashing on app launch with the following output:
> Cannot create property 'runCommand' on string 'LottieAnimationView' 

I've also had no luck attempting to disable strict mode via Typescript or Babel (and I'm targeting hermes)

**Comment by tido64** [comment, marked as answer]
Unfortunately, we currently do not have any plans to address this. Since esbuild _always_ adds the directive when not targeting ESM, manually removing it means that we also have to adjust all the sourcemap offsets. If this is something you want to look into, we'd be happy to accept PRs.

That said, apps breaking with `"use strict";` tend to be due to real bugs. In this case, one of Lottie's dependencies, `react-native-safe-module`, is making assumptions it shouldn't have. I can see that it was [removed](https://github.com/lottie-react-native/lottie-react-native/commit/5bff05c159852ecd06d9a0206dc525c18e8a07f9) in a more recent version. If you can upgrade to 6.x, it should resolve the crash and also set you on a path to enable New Architecture in the future.
